### PR TITLE
Change runtime to bun

### PR DESCRIPTION
**Pull request by KaiVolland**
This changes the runtime and creation of the binaries from node to [bun](https://bun.sh/).

It also adds options to read ArcGIS styles via [geostyler-lyrx-parser](https://github.com/geostyler/geostyler-lyrx-parser).

### :warning:  As this changes the runtime of this project this will result in a new major version.

### CODE REVIEW

**Comment by marcjansen** [comment]
Are the changes to the installation guide (or the part usage without installation) in the `README` needed? It will still be possibly to run this CLI without `bun`, or not?

File: package.json [hunk 1]
```diff
@@ -68,16 +73,17 @@
     "typescript": "^5.0.4"
   },
   "dependencies": {
-    "geostyler-mapbox-parser": "^5.0.1",
-    "geostyler-mapfile-parser": "^3.0.0",
-    "geostyler-openlayers-parser": "^4.3.0",
-    "geostyler-qgis-parser": "^2.0.1",
-    "geostyler-sld-parser": "^5.3.1",
-    "geostyler-style": "^8.1.0",
+    "geostyler-lyrx-parser": "^1.0.0",
```
Comment (ger-benjamin) [inline package.json:76]: May I ask you to take the version `1.0.1` directly ?

Comment (ger-benjamin) [reply package.json]: Thanks!

**Comment by KaiVolland** [comment]
> Are the changes to the installation guide (or the part usage without installation) in the `README` needed? It will still be possibly to run this CLI without `bun`, or not?

I was not able to get the module resolution working with `node` and the current state of the geostyler packages. For some reason node has issues resolving relative pathes in the node_modules. `bun` seems to be smarter here.

I also struggle validating `npx` and `bunx` without publishing the package. :thinking:

**Review by geographika — APPROVED** [review]
I only learnt of bun at the codesprint, so I've no useful opinion on switching from one to the other. However, if it addresses https://github.com/geostyler/geostyler-cli/issues/393 then I am all for it!

**Comment by jansule** [comment]
You also might want to rephrase your commit messages so that the support for openlayers does not appear in the changelog

File: src/logHelper.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import gradient from 'gradient-string';
-import version from './version';
+import { version } from '../package.json';
```
[the remark this answers is not in the document]
Comment (KaiVolland) [reply src/logHelper.ts:2]: This works as expected when using bun.

**Comment by marcjansen** [comment]
> > Are the changes to the installation guide (or the part usage without installation) in the `README` needed? It will still be possibly to run this CLI without `bun`, or not?
> 
> I was not able to get the module resolution working with `node` and the current state of the geostyler packages. For some reason node has issues resolving relative pathes in the node_modules. `bun` seems to be smarter here.
> 
> I also struggle validating `npx` and `bunx` without publishing the package. 🤔

I am +0 on the change of the runtime, so not blocking.

I really would like to stress that installation via node / npx should still work even after we swithc to using it. Maybe you can ctreate a prerelease and test this?

**Comment by KaiVolland** [comment]
> > > Are the changes to the installation guide (or the part usage without installation) in the `README` needed? It will still be possibly to run this CLI without `bun`, or not?
> > 
> > 
> > I was not able to get the module resolution working with `node` and the current state of the geostyler packages. For some reason node has issues resolving relative pathes in the node_modules. `bun` seems to be smarter here.
> > I also struggle validating `npx` and `bunx` without publishing the package. 🤔
> 
> I am +0 on the change of the runtime, so not blocking.
> 
> I really would like to stress that installation via `node` / `npx` should still work even after we switch to no longer using or endorsing them. Maybe you can create a prerelease and test this?

After this change, `node' and `npx' will no longer work.

**Comment by ahocevar** [comment]
Isn't this a change more radical than it needs to be? Using bun to create the standalone binaries is fine. But what's the benefit of getting rid of a build that works with node? I just created #407, by creating a single javascript file that runs fine everywhere, using rollup. There's no need to get rid of node support, while still being able to use bun.

**Comment by KaiVolland** [comment]
This is superseeded by #407. I'll create a new PR with the changes that are still useful later.

**Pull request closed without merging**